app/frontend/components/pages/delegations/delegatePage.tsx

```diff
@@ -46,58 +49,145 @@ const BigDelegatorOffer = (): h.JSX.Element => (
-  txSuccessTab,
-  gettingPoolInfo,
-  poolRecommendation,
-  pool,
-  isBigDelegator,
-  withAccordion,
-  title,
-  confirmTransaction,
-}: Props): h.JSX.Element => {
-  /*
-  REFACTOR (calculateFee):
-  Calculation of "fee" should be fired from there explicitely using "useEffect"
-  when component mounts.
-  */
+const Delegate = ({withAccordion, title}: Props): h.JSX.Element => {
+  const {
+    txSuccessTab,
+    stakePool,
+    currentDelegation,
+    calculatingDelegationFee,
+    delegationFee,
+    delegationValidationError,
+    gettingPoolInfo,
+    isBigDelegator,
+    isShelleyCompatible,
+    poolRecommendation,
+    validStakepoolDataProvider,
+  } = useSelector((state) => ({
+    // REFACTOR: (Untyped errors)
+    delegationValidationError: state.delegationValidationError,
+    stakePool: state.shelleyDelegation.selectedPool,
+    currentDelegation: getSourceAccountInfo(state).shelleyAccountInfo.delegation,
+    calculatingDelegationFee: state.calculatingDelegationFee,
+    delegationFee: state.shelleyDelegation.delegationFee,
+    txSuccessTab: state.txSuccessTab,
+    gettingPoolInfo: state.gettingPoolInfo,
+    isShelleyCompatible: state.isShelleyCompatible,
+    poolRecommendation: getSourceAccountInfo(state).poolRecommendation,
+    isBigDelegator: isBigDelegatorSelector(state),
+    validStakepoolDataProvider: state.validStakepoolDataProvider,
+  }))
+  const {confirmTransaction, updateStakePoolIdentifier} = useActions(actions)
+
+  // TODO: this reference always changes inside "useEffect" which causes infite loop.
+  // Till this is investigated, this workaround is used instead of "eslint-disable-..."
+  const _updateStakePoolIdentifier = useRef(updateStakePoolIdentifier)
+
+  const [fieldValue, setFieldValue] = useState('')
+  const [error, setError] = useState<Error>(null)
+
+  const handleInputValidation = useCallback(
+    (value: string) => {
+      const {poolHash, error} = validateInput(value, validStakepoolDataProvider)
+      if (!error) {
+        /*
+      Relates to REFACTOR (calculateFee):
+      This component should store all `txPlan` and `error` states & render confirmation modal
+      when user clicks on "Delegate". This could greatly simplify global state flow. For now
+      using hybrid solution involving `updateStakePoolIdentifier` (should be later removed)
+      */
+        _updateStakePoolIdentifier.current(poolHash)
+        setError(null)
+      } else {
+        setError(error)
+      }
+    },
+    [validStakepoolDataProvider]
+  )
+
+  const handleOnInput = (event: any): void => {
+    const newValue: string = event?.target?.value
+    setFieldValue(newValue)
+
+    // Note: this could be debounced
+    if (newValue) {
+      handleInputValidation(newValue)
+    }
+  }
+
+  // init "stake pool input" and refresh it when "currentDelegation" changes
+  useEffect(() => {
+    const recommendedPoolHash =
+      poolRecommendation?.recommendedPoolHash || currentDelegation?.poolHash
```

Comment:
Do we need the currentDelegation here just so the effect is triggered when its changed? Because otherwise it doesn't make a lot of sense to me to set `recommnededPoolHash` to currentDelegation.poolHash

Reply:
Yes I need it there because of that and this `poolRecommendation?.recommendedPoolHash || currentDelegation?.poolHash` condition was there before so I did not want to change it. Should I change it just to `poolRecommendation?.recommendedPoolHash`? I can then still keep `currentDelegation` in the dependencies of `useEffect`.

Reply:
Lets keep it as it is, we should address the pool recommendation in another PR.